src/psyclone/psyir/transformations/intrinsics/sum2code_trans.py

```diff
@@ -0,0 +1,361 @@
+                # Exception handled by validate method
+                pass
+            ndims = len(array_ref.symbol.shape)
+
+            loop_bounds = []
+            for idx, shape in enumerate(array_ref.symbol.shape):
+                if shape in [ArrayType.Extent.DEFERRED,
+                             ArrayType.Extent.ATTRIBUTE]:
+                    # runtime extent using LBOUND and UBOUND required
+                    lbound = BinaryOperation.create(
+                        BinaryOperation.Operator.LBOUND,
+                        Reference(array_ref.symbol),
+                        Literal(str(idx+1), INTEGER_TYPE))
+                    ubound = BinaryOperation.create(
+                        BinaryOperation.Operator.UBOUND,
+                        Reference(array_ref.symbol),
+                        Literal(str(idx+1), INTEGER_TYPE))
+                    loop_bounds.append((lbound, ubound))
+                elif isinstance(shape, ArrayType.ArrayBounds):
+                    # array extent is defined in the array declaration
+                    loop_bounds.append(shape)
+                #  else: Exception is handled by validate method
+        else:
+            # This is an array reference
+            loop_bounds = []
+            ndims = len(array_ref.children)
+            for shape in array_ref.children:
+                loop_bounds.append((shape.start.copy(), shape.stop.copy()))
+
+        # Determine the datatype of the array's values and create a
+        # scalar of that type
+        array_intrinsic = array_ref.symbol.datatype.intrinsic
+        array_precision = array_ref.symbol.datatype.precision
+        scalar_type = ScalarType(array_intrinsic, array_precision)
+
+        symbol_table = node.scope.symbol_table
+        assignment = node.ancestor(Assignment)
+
+        datatype = scalar_type
+        array_reduction = False
+        if dimension_ref and ndims > 1:
+            array_reduction = True
+            # We are reducing from one array to another
+            shape = []
+            for idx, bounds in enumerate(loop_bounds):
+                if int(dimension_literal.value)-1 == idx:
+                    pass
+                else:
+                    shape.append(bounds)
+            datatype = ArrayType(scalar_type, shape)
+
+        # Create temporary sum variable (sum_var)
+        symbol_sum_var = symbol_table.new_symbol(
+            "sum_var", symbol_type=DataSymbol, datatype=datatype)
+
+        # Replace operation with a temporary variable (sum_var).
+        node.replace_with(Reference(symbol_sum_var))
+
+        # sum_var=0.0 or 0
+        lhs = Reference(symbol_sum_var)
+        if scalar_type.intrinsic == scalar_type.Intrinsic.REAL:
+            rhs = Literal("0.0", scalar_type)
+        elif scalar_type.intrinsic == scalar_type.Intrinsic.INTEGER:
+            rhs = Literal("0", scalar_type)
+        else:
+            # exception handled by validate method
+            pass
```

Comment:
Remove else/pass and leave it as a comment

Reply:
Added comment.